Feat: Add compact workspace drawer style

DenysMb opened the pull request:
<!-- Thank you for your Pull Request. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!-- Please start by naming your pull request properly for e.g. "Add Google Tasks to Todo providers". -->
<!-- Please keep in mind that any text inside "<!--" and "--\>" are comments from us and won't be visible in your bug report, so please don't put any text in them. -->

#### Pre-flight Checklist

Please ensure you've completed all of the following.

- [X] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/HEAD/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that this project adheres to.

#### Description of Change
<!-- Describe your changes in detail. -->
Added an option "Use compact workspace drawer". This option shows a smaller version of the workspace drawer (the same width as the sidebar), displaying the "All services" workspace as displaying the first character of the name of all the user-created workspace (so if we start the name of the workspace with an emoji, for example, we'll display the emoji).
I always fixed a bug 

#### Motivation and Context
<!-- Why is this change required? What problem does it solve?  If it fixes an open issue, please link to the issue here. -->
The workspace drawer is too big to use the "Always show workspace drawer" option comfortably.
In order to fix this, I added a option to use a compact workspace drawer style.

#### Screenshots
<!-- Remove the section if this does not apply. -->
<img width="1504" height="997" alt="Screenshot_20251122_202016" src="https://github.com/user-attachments/assets/729a59c7-0ac9-4a1d-9965-b5e8e76ef883" />
<img width="1504" height="997" alt="Screenshot_20251122_202028" src="https://github.com/user-attachments/assets/23455b21-2365-478e-9275-00086ebedeb9" />
<img width="1504" height="997" alt="Screenshot_20251122_202045" src="https://github.com/user-attachments/assets/56ad4c0b-fbdf-43e9-b95c-cabd8afb2d71" />

https://github.com/user-attachments/assets/4ff939aa-e9f8-4e84-961b-95fcbd63c29b

https://github.com/user-attachments/assets/e3953b3a-58ba-4a71-a621-999ee523af14



#### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [X] My pull request is properly named
- [X] The changes respect the code style of the project (`pnpm prepare-code`)
- [X] `pnpm test` passes
- [X] I tested/previewed my changes locally

#### Release Notes

<!-- Please add a one-line description for users of Ferdium to read in the release notes, or 'none' if no notes relevant to such users. Examples and help on special cases: https://github.com/electron/clerk/blob/master/README.md#examples -->
Added "Use compact workspace drawer" option.

DenysMb committed:
Feat: Add button for compact workspace drawer mode

DenysMb committed:
Feat: Add compact workspace styles

DenysMb committed:
Fix: Fix lint problems

DenysMb committed:
Feat: Better compact workspace styles

DenysMb committed:
Feat: Use only first letter on compact workspace mode

DenysMb committed:
Feat: Add workspace list to one div

DenysMb committed:
Feat: Add workspace indicator

DenysMb committed:
Feat: Fix changes for non-compact mode

DenysMb committed:
Feat: Remove added borders

DenysMb committed:
feat: match compact workspace items height with sidebar items height

DenysMb committed:
feat: remove workspace item to display service name

DenysMb committed:
feat: match compmact workspace drawer width with sidebar width

DenysMb committed:
fix: apply workspace drawer width calculation when changing sidebar
width

DenysMb committed:
feat: track ribbon width changes separately and temporarily disable CSS
transitions to prevent visual glitches

DenysMb committed:
fix: prevent workspace drawer to appear on the right of window when
hidden

DenysMb committed:
feat: update workspace drawer transform logic to respect the new
alwaysShowWorkspaces flag

DenysMb committed:
feat: improve drawer animation transitions

DenysMb committed:
fix: prevents visual glitches when the drawer is open or "always show"
is enabled.

DenysMb committed:
feat: add workspace icon to compact drawer items

DenysMb committed:
feat: respect reduced motion preference in workspace animations

DenysMb committed:
feat: add compact workspace drawer settings text to locale file

tbaumann commented:
I love this. Switching workplaces takes so many clicks right now. (keeping the drawer open is not an option at it's current size)

DenysMb committed:
Fix: Address Copilot review suggestions for compact workspace drawer

- Fix matchMedia always-truthy bug in generateAppContentTransition:
  window?.matchMedia?.(query) returns a MediaQueryList (truthy) even
  when the media query does not match; use reducedMotionQuery?.matches
  to evaluate the actual match result

- Normalize settings to appSettings in updateStyle to prevent
  updateProgressbar receiving the full settings store (which lacks
  progressbarAccentColor at the top level), fixing broken progress
  bar coloring when called from the new drawer-settings reaction

- Guard getDrawerAnimationDuration against non-browser environments
  with typeof window check before calling window.matchMedia

- Rename compactClass prop to isCompact: boolean in WorkspaceDrawerItem
  to remove dual-purpose string/boolean usage; derive the CSS class
  name locally from the boolean flag

- Add aria-label={name} to compact workspace items so screen readers
  and assistive tech can identify items when the visual label is
  reduced to an icon/initial

- Use CSS variable var(--workspace-drawer-width) as single source of
  truth for drawer width in WorkspaceDrawer JSS styles, removing the
  hard-coded theme values and the compact override that could cause
  inconsistencies with the appearance-injected CSS

emilthemaker commented:
This is a great PR, solves a problem I've been running into. Putting up a $25 bounty for you + $15 for whoever merges this into the main project as a thank you if this gets merged in (must have Github sponsors or sth similar) @kurobeats

DenysMb committed:
Merge branch 'develop' into develop

DenysMb commented:
> This is a great PR, solves a problem I've been running into. Putting up a $25 bounty for you + $15 for whoever merges this into the main project as a thank you if this gets merged in (must have Github sponsors or sth similar)

I worked on the comments, everything is working fine, I think now we just need to wait...

SpecialAro reviewed (approved):
LGTM! Thank you for your work on this! 😄

SpecialAro commented:
@all-contributors please add @DenysMb for code
